### PR TITLE
Initial version of VerneMQ HTTP API

### PR DESCRIPTION
**Pull request by dergraf**
The idea is that the HTTP API wrapps all commands of the vmq-admin CLI tool.

https://asciinema.org/a/by17s5dpitqi7qp1blm7zh2bx

### CODE REVIEW

**Comment by dergraf** [comment]
@larshesel please review

**Comment by larshesel** [comment]
Hi @dergraf I'll get to this tomorrow or the day after! Haven't forgotten about it... Xmas and all ;)

File: apps/vmq_server/src/vmq_http_mgmt_api.erl [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+rest_init(Req, _Opts) ->
+    {ok, Req, undefined}.
+
+allowed_methods(Req, State) ->
+    {[<<"POST">>, <<"GET">>, <<"OPTIONS">>, <<"HEAD">>], Req, State}.
+
+content_types_provided(Req, State) ->
+    {[{<<"application/json">>, to_json},
+      {<<"text/html">>, to_html}], Req, State}.
```
Comment (larshesel) [inline apps/vmq_server/src/vmq_http_mgmt_api.erl]: Why the second content type. Won't all the content returned be strictly JSON? As the `to_html/2` function just calls the `to_json/2` function I suppose everything really will be JSON and not HTML?

File: apps/vmq_server/priv/vmq_server.schema [hunk 2]
```diff
@@ -577,26 +590,25 @@
...
-%% @doc listener.http.metrics is the IP address and TCP port that
-%% the broker will bind to accept HTTP connections from metrics collectors
-%% such as Prometheus e.g:
-%% - listener.http.metrics = 0.0.0.0:8888
+%% @doc listener.http.default is the IP address and TCP port that
+%% the broker will bind to accept HTTP connections
+%% - listener.http.default = 0.0.0.0:8888
 %% This also works for SSL listeners:
-%% - listener.https.metrics= 0.0.0.0:8889
+%% - listener.https.default= 0.0.0.0:8889
```
Comment (dergraf) [inline apps/vmq_server/priv/vmq_server.schema]: exactly

File: apps/vmq_server/src/vmq_http_mgmt_api.erl (hunk 1 above)
Comment (dergraf) [inline apps/vmq_server/src/vmq_http_mgmt_api.erl]: Hmm not sure why I added it in the first place... But you're right, application/json should be enough

File: apps/vmq_server/src/vmq_http_mgmt_api.erl [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+%%% Cowboy Config
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+routes() ->
+    [{"/api/v1/[...]", ?MODULE, []}].
+
+create_api_key() ->
+    Chars = list_to_tuple("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"),
+    Size = size(Chars),
+    F = fun(_, R) -> [element(rnd:uniform(Size), Chars) | R] end,
+    ApiKey = list_to_binary(lists:foldl(F, "", lists:seq(1, 32))),
```
Comment (dergraf) [inline apps/vmq_server/src/vmq_http_mgmt_api.erl]: you're right, it was a copy paster.

**Comment by larshesel** [comment]
@dergraf I just pushed the changes for this - feel free to squash/merge if you're ok with the changes...

**Pull request merged**